src/app/screens/connectors/ConnectBtcpayCln/index.tsx

```diff
@@ -0,0 +1,136 @@
+import CompanionDownloadInfo from "@components/CompanionDownloadInfo";
+import ConnectorForm from "@components/ConnectorForm";
+import TextField from "@components/form/TextField";
+import ConnectionErrorToast from "@components/toasts/ConnectionErrorToast";
+import axios from "axios";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+import msg from "~/common/lib/msg";
+
+const initialFormData = {
+  url: "",
+  macaroon: "",
+  name: "",
+};
+
+export default function ConnectBtcpayCln() {
+  const navigate = useNavigate();
+  const { t } = useTranslation("translation", {
+    keyPrefix: "choose_connector.btcpay_cln",
+  });
+  const [formData, setFormData] = useState(initialFormData);
+  const [loading, setLoading] = useState(false);
+  const [hasTorSupport, setHasTorSupport] = useState(false);
+
+  function getConfigUrl(data: string) {
+    const configUrl = data.trim().replace("config=", "");
+    try {
+      return new URL(configUrl);
+    } catch (e) {
+      return null;
+    }
+  }
+  async function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const configUrl = getConfigUrl(event.target.value);
+    if (!configUrl) {
+      return;
+    }
+    const host = configUrl.host;
+    try {
+      const response = await axios.get<{
+        configurations: [{ uri: string; adminMacaroon: string }];
+      }>(configUrl.toString());
+
+      if (response.data.configurations[0].uri) {
+        setFormData({
+          url: response.data.configurations[0].uri,
+          macaroon: response.data.configurations[0].adminMacaroon,
+          name: host,
+        });
+      }
+    } catch (e) {
+      console.error(e);
+      toast.error(t("errors.connection_failed"));
+    }
+  }
+
+  function getConnectorType() {
+    if (formData.url.match(/\.onion/i) && !hasTorSupport) {
+      return "nativelnd";
+    }
+    // default to LND
+    return "lnd";
+  }
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    setLoading(true);
+    const { url, macaroon, name } = formData;
+    const account = {
+      name: name || "LND",
```

Comment:
@pavanjoshi914 @reneaaron not sure about this part of the code: `name:` It needs to be changed?